test/fw/ptl/lib/ptl_comm.py

```diff
@@ -0,0 +1,270 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2020 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of both the OpenPBS software ("OpenPBS")
+# and the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# OpenPBS is free software. You can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version.
+#
+# OpenPBS is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public
+# License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# PBS Pro is commercially licensed software that shares a common core with
+# the OpenPBS software.  For a copy of the commercial license terms and
+# conditions, go to: (http://www.pbspro.com/agreement.html) or contact the
+# Altair Legal Department.
+#
+# Altair's dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of OpenPBS and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair's trademarks, including but not limited to "PBS™",
+# "OpenPBS®", "PBS Professional®", and "PBS Pro™" and Altair's logos is
+# subject to Altair's trademark licensing policies.
+
+
+import copy
+import datetime
+import logging
+import os
+import re
+import socket
+import string
+import sys
+import time
+try:
+    from ptl.lib.pbs_ifl import *
+    API_OK = True
+except:
+    try:
+        from ptl.lib.pbs_ifl_mock import *
+    except:
+        sys.stderr.write("failed to import pbs_ifl, run pbs_swigify " +
+                         "to make it\n")
+        raise ImportError
+    API_OK = False
+
+
+from ptl.lib.pbs_testlib import *
+
+def get_comm_obj(name=None, attrs={}, pbsconf_file=None, snapmap={},
```

Comment:
Make server arg as mandatory in each of daemon class then we don't need this method.